File: frontend/src/lib/api/schema.ts
import { GetDatabases, GetSchemas, GetTables, GetViews, GetColumns } from '$wailsjs/go/bindings/SchemaService'

export type Column = {
  Name: string
  Type: string
  Nullable: boolean
  Default: string
}

export async function getDatabases(connectionId: string): Promise<string[]> {
  return GetDatabases(connectionId)
}

export async function getSchemas(connectionId: string, database: string): Promise<string[]> {
  return GetSchemas(connectionId, database)
}

export async function getTables(connectionId: string, schemaName: string): Promise<string[]> {
  return GetTables(connectionId, schemaName)
}

export async function getViews(connectionId: string, schemaName: string): Promise<string[]> {
  return GetViews(connectionId, schemaName)
}

export async function getColumns(connectionId: string, schemaName: string, table: string): Promise<Column[]> {
  return GetColumns(connectionId, schemaName, table)
}

File: internal/schema/schema.go
package schema

type Inspector interface {
	GetDatabases() ([]string, error)
	GetSchemas(database string) ([]string, error)
	GetTables(schema string) ([]string, error)
	GetViews(schema string) ([]string, error)
	GetColumns(schema, table string) ([]Column, error)
}

type Column struct {
	Name     string
	Type     string
	Nullable bool
	Default  string
}
